File: src/main/java/com/ftn/ues/email_client/dao/elastic_search/Tag.java
package com.ftn.ues.email_client.dao.elastic_search;

import com.ftn.ues.email_client.dao.DirectMapping;
import com.ftn.ues.email_client.model.User;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.experimental.SuperBuilder;

@Data
@AllArgsConstructor
@SuperBuilder
public class Tag extends DirectMapping<com.ftn.ues.email_client.model.Tag> {

    private Long id;
    private String name;
    private Long user;

    @Builder.Default
    private Boolean deleted = false;

    public Tag(com.ftn.ues.email_client.model.Tag object) {
        super(object);
        id = object.getId();
        name = object.getName();
        user = object.getUser().getId();
        deleted = object.getDeleted();
    }

    @Override
    public com.ftn.ues.email_client.model.Tag getModelObject() {
        return com.ftn.ues.email_client.model.Tag.builder()
                .id(id)
                .name(name)
                .user(User.builder().id(user).build())
                .deleted(deleted)
                .build();
    }
}

File: src/main/scala/com/ftn/ues/email_client/dao/elastic/Tag.scala
package com.ftn.ues.email_client.dao.elastic

import play.api.libs.json.{Json, OFormat}

case class Tag(id: Long, name: String, userId: Long, deleted: Boolean)

object Tag {
  def apply(tag: com.ftn.ues.email_client.model.Tag): Tag =
    Tag(
      tag.getId,
      tag.getName,
      tag.getUser.getId,
      tag.getDeleted
    )

  implicit val format: OFormat[Tag] = Json.format[Tag]
}
